### DISCUSSION TITLE
virus/malware - Analysis Overview

### DISCUSSION THREAD

**Discussion by thezoggy** [sabnzbd/sabnzbd#2870] [General]
Saw a thing about hybrid-analysis and figured i'd try it out with latest sab installer to see what it says.

Of course due to the nature of sab and the whole python app in a exe, always makes malware/antivirus stuff unhappy. But was shocked to see it was pretty good with most things.

### file: SABnzbd-4.3.2RC2-win-setup.exe 
os: Windows 10 64 bit

**results:**
http://hybrid-analysis.com/sample/001b8c18b10ebff66c00380b02658f8c5e3686360886cf89f608e3116479f9fd

Can see the malicious reason is the whole launching readme.txt

Which is only seen by the analysis because we default that option during installer.

I don't know if we want to just default that to uncheck by default?
Or perhaps as an alternative we should just stop doing that and instead show link for readme to website / offer launching link to it?

---

Tried "SABnzbd-4.3.2RC2-osx.dmg" w/Mac Catalina 64 bit (x86) - but it shows "Failed unpacking archive, unknown error"

**Comment by thezoggy** [comment]
had a few people hit up irc/discord that they cant download the windows setup with their browser..  tried myself and i see ms edge (microsoft defender smartscreen) blocks it because it hasnt been trusted/seen.  one of the triggers is that the publisher is unknown..  

i could have sworn we used to populate this info.. but looks like we dont:
![explorer_2024-06-14_12-01-25](https://github.com/sabnzbd/sabnzbd/assets/500882/3ed571aa-2353-4a9a-a62f-36ac031f8d1e)

just on the app inside,
![explorer_2024-06-14_12-03-55](https://github.com/sabnzbd/sabnzbd/assets/500882/8e2324e0-c784-484f-8aa8-2452d2240e89)


so we should deff go add that data

**Reply by Safihre** [reply]
Fixed this. But doubt it will help much.

**Reply by mnightingale** [reply]
Have you looked into code signing options recently?

Microsoft have a new [Trusted Signing](https://learn.microsoft.com/en-us/azure/trusted-signing/overview) service, I'm not familiar with the details for instance if the organisation is required to be a legal entity or if you just need to create a tenant and add payment details to Azure. There are actions to make it simple to integrate with GitHub https://github.com/Azure/trusted-signing-action

There are also Open Source Code Signing certificates https://shop.certum.eu/open-source-code-signing.html but it's probably more tricky to integrate with a GitHub pipeline.

**Reply by Safihre** [reply]
> There are also Open Source Code Signing certificates https://shop.certum.eu/open-source-code-signing.html but it's probably more tricky to integrate with a GitHub pipeline.

I have checked this out and bought it. But it requires me to be personally be listed on the signature. They do a proper organization and person check.
Which of course makes sense.. Can't be handing out certificates to random internet peepz 😂
We have a business registration for PayPal, but it's also registered to shypike's address so I doubt he wants to list that.

**Reply by mnightingale** [reply]
I came across https://signpath.org [Documentation for GitHub](https://about.signpath.io/documentation/trusted-build-systems/github) approval is a manual step, so it may only be useful for final releases. You can also only sign your own code, if that matters?

**Reply by Safihre** [reply]
@mnightingale I have applied for Signpath and was approved. 
I [implemented](https://github.com/sabnzbd/sabnzbd/tree/refs/heads/feature/signpath) the signing and now awaiting for their final approval before they issue me the production certificate.
Was much easier than expected!

**Reply by Safihre** [reply]
We got the approval and tested the build with the release certificates:
https://github.com/sabnzbd/sabnzbd/actions/runs/17868921805
Both installer and the binaries are signed.

It does show the publisher as `SignPath Foundation` in the "Are you sure you want to run this?" pop-up.
But I think I will reference that on the Downloads page and in the Readme.
They require in any case that we acknowledge them on our repo (understandably), so that would then satisfy both criteria.

I initially also signed the binaries of 7zip and par2-cmdline with the test-setup, but this is not allowed. We are only allowed to signed our "own" code.
Reference: https://signpath.org/terms

Will do the text updates in the upcoming days.

**Comment by Safihre** [comment]
So, I released 4.5.4Beta1 with the code signing.
However, Defender and my anti-virus still don't trust it because it us "uncommon".
Wonder if this really helps at all..

**Reply by thezoggy** [reply]
grabbing  SABnzbd-4.5.4Beta1-win-setup.exe 

the sab installer i do see signed now:
<img width="820" height="506" alt="image" src="https://github.com/user-attachments/assets/b1536632-dc15-4e61-bf60-fb11db13b2d1" />

same with the sab binary:
<img width="829" height="556" alt="image" src="https://github.com/user-attachments/assets/1d1102a8-8db3-421c-9d43-58a40a1f88db" />

so hopefully things will be happier.

**Reply by mnightingale** [reply]
@Safihre Have you done the 'Report this file as safe' step?
I don't know if it will improve over time, maybe with trust in releases with the same signature increasing.
Perhaps it being hosted on GitHub also has an impact.
I'd suggest, reporting it as safe, see if if it stops flagging, then see what happens with the next exe (stable or beta 2).

**Reply by thezoggy** [reply]
I know I've submitted stuff in the past (like I'm sure others do) and after a few days/weeks it usually has enough rep that people dont get hit anymore. Which is why saf usually delays windows releases purposely on the rss feed so that happens

**Reply by Safihre** [reply]
Yeah, I was just hoping that wouldn't be needed anymore.
Basically on my system I don't see any difference between how Edge and Windows respond to the signed 4.5.4 or the unsigned 4.5.3.
So why even bother...?
The release being served by github.com might be an issue I supposed?
@mnightingale you saw somewhere that the general certificate becomes trusted after some time? Or it's a hope?😁

**Reply by mnightingale** [reply]
Edge isn’t warning me anymore, is that quicker than usual?
Mostly hope, but afaik it’s all part of the SmartScreen system, I would expect amongst other things it scores reputation based on the URL of the download and the digital signature.
Since this the first time it has seen the signature, and it appears to have stopped warning relatively quickly I’d stick with it and see if over time as reputation builds it warns less or for an even shorter amount of time.

**Reply by Safihre** [reply]
Seems you were right. For 4.5.4Beta2 my Edge accepted the binary right away without the extra warning! 🎉
